db/compaction/compaction_job_test.cc

```diff
@@ -374,7 +374,7 @@ class CompactionJobTestBase : public testing::Test {
     } else if (table_type_ == TableTypeForTest::kMockTable) {
       file_size = 10;
       EXPECT_OK(mock_table_factory_->CreateMockTable(
-          env_, GenerateFileName(file_number), std::move(contents)));
```

Comment:
The compiler complained about a redundant move, so I assumed it was right.

Reply:
It seems 'contents' is a const reference, so there is no point in moving it into a value.

Reply:
https://github.com/rockset/rocksdb-cloud/blob/master/table/mock_table.h#L68 weird it complains.
seems `contents` here is a value

Reply:
My understanding of the problem is as follows. Inside this Method (AddMockFile) contents is a const reference (void AddMockFile(const mock::KVVector& contents, int level = 0)) so when we pass the reference to CreateMockTable which takes an object we need to call the copy constructor to construct the object. If we call std::move(contents) then we force the compiler to first create the object by calling the copy constructor and then call the move constructor to move it to the argument. If we remove the std::move we are just going to construct the object in place for the argument.

Reply:
oh I see what your mean, yeah my bad. Thanks for explaining

Reply:
@dpetrov4 let's please try very hard to avoid changes to our fork that exist in upstream. Upgrading RocksDB is already extremely painful, and while each of these small fixes won't be hard to merge individually, they do add up. If the compiler complains let's turn off warnings for this file or fix upstream: https://github.com/facebook/rocksdb/blob/main/db/compaction/compaction_job_test.cc#L381